src/client/components/questions/Address.jsx

```diff
@@ -83,23 +94,56 @@ class Address extends React.Component {
             value={this.props.value.country}
             required={this.props.required}
             onValueChange={(update) => {this.handleChange('country', update);}}/>
+      </div>
+      );
 
-        <ErrorableSelect errorMessage={this.validateAddressField(this.props.value.state) ? undefined : 'Please enter a valid state'}
-            label="State"
-            name="state"
-            autocomplete="address-level1"
-            options={stateList}
-            value={this.props.value.state}
-            required={this.props.required}
-            onValueChange={(update) => {this.handleChange('state', update);}}/>
+    let secondaryForm = null;
+    if (this.state.requireStateZip) {
+      secondaryForm = (
+        <div>
+          <ErrorableSelect errorMessage={this.validateAddressField(this.props.value.state) ? undefined : 'Please enter a valid state'}
+              label="State/Province"
+              name="state"
+              autocomplete="address-level1"
+              options={stateList}
+              value={this.props.value.state}
+              required={this.state.requireStateZip ? this.props.required : false}
```

Comment:
this conditional will always be true as this is in a block dependent on the condition being true L101